frontend/src/metabase/actions/containers/ActionCreator/ActionCreator.unit.spec.tsx

```diff
@@ -58,11 +77,24 @@ async function setup({ action }: SetupOpts = {}) {
 async function setupEditing({
   action = createMockQueryAction(),
   ...opts
-} = {}) {
+}: SetupOpts = {}) {
   await setup({ action, ...opts });
   return { action };
 }
 
+const DATA_REFERENCE_ICON_LABEL = "reference icon";
+const ACTION_SETTINGS_ICON_LABEL = "gear icon";
```

Comment:
Think it'd be better to use `getIcon` / `queryIcon` from `__support__/ui`:

```jsx
import { getIcon, queryIcon } from "__support__/ui"

expect(getIcon("reference")).toBeInTheDocument()
expect(queryIcon("gear")).not.toBeInTheDocument()
```

Reply:
Didn't know these exists too.

Reply:
They're pretty new!

Reply:
unfortunately this doesn't work. The util return just the icon, but it's not clickable. I have a few options
1. keep using what's defined here
1. create another 2 util funtions i.e. `getIconButton` and `queryIconButton`
1. use `getIcon("reference").closest("button")` which kind of defeats the purpose of using a util function since we still need to manipulate it somehow.

I think 2. is ideal. What do you think?

Reply:
Oh, I see; thanks for giving it a shot though. I don't have a solid opinion on that: some icon buttons are implemented as HTML buttons, and some use the `a` tag. A future with `getIcon`, `getIconButton`, and `getIconLink` makes me weep to be honest, so I'd stick to 1 🤷 Also, we can consider making `getIcon` / `queryIcon` a bit smarter by adding a `role` argument. What do you think about something like that?

```tsx
function getIcon(iconName: string, role?: string) {
  const name = `${iconName} icon`;
  return role
    ? screen.getByRole(role, { name })
    : screen.getByLabelText(name);
}
```

Reply:
sounds good. I like that better than having 2 more similar implementations. I'll give it another shot, my first thought is that getByRole and getByLabelText is too different and imo shouldn't be in the same function, I think I can improve this a bit but gotta try.